### PR TITLE
S4 t1 put route

### PR DESCRIPTION
**Pull request by luke-grayland**

### CODE REVIEW

File: Config/Routes.js [hunk 1]
```diff
@@ -1,10 +1,20 @@
-const ProductController = require("../Controllers/ProductController");
+const DbService = require('../Services/DbService')
+const ProductController= require("../Controllers/ProductController");
 
 function routes(app) {
     app.get('/products', ProductController.getAllProducts)
     app.post('/products', ProductController.addSingleProduct)
+    app.put('/products', ProductController.updateSingleProduct)
```
Comment (Parsley-Afroman) [inline Config/Routes.js:7]: should this not also be invalid as the 'updateSingleProduct' requires the 'SKU' in the URL and has a route on line 11

File: Config/Routes.js [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
+    app.put('/products', ProductController.invalidRoute)
```
Comment (Parsley-Afroman) [inline Config/Routes.js:8]: should we set up delete pathways as invalid routes as well?

File: Config/Routes.js [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
+
     app.get('/products/:SKU', ProductController.getSingleProduct)
-    app.put('/products/:SKU', ProductController.updateStockLevel)
+    app.put('/products/:SKU', ProductController.updateSingleProduct)
+    app.post('/products/:SKU', ProductController.invalidRoute)
```
Comment (Parsley-Afroman) [inline Config/Routes.js:12]: a delete pathway here with an invalid route function?

File: Controllers/ProductController.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 const DbService = require("../Services/DbService");
 const ProductService = require("../Services/ProductsService");
 const JsonResponseService = require("../Services/JsonResponseService");
+const {sanitiseSku, sanitiseName, sanitiseStockLevel, sanitisePrice} = require("../Services/SanitiseValidateService");
```
Comment (Parsley-Afroman) [inline Controllers/ProductController.js:4]: Don't need this line as it is facilitated by the one below and prepending the the function with 'DataCheckers.'

File: Controllers/ProductController.js [hunk 5]
```diff
@@ -13,6 +14,7 @@ const getAllProducts = async(req, res) => {
     res.json(apiResponse)
 }
 
+
```
Comment (Parsley-Afroman) [inline Controllers/ProductController.js:17]: remove space

File: package.json [hunk 6]
```diff
@@ -24,3 +24,5 @@
     "jest": "^28.0.1"
   }
 }
+
```
Comment (Parsley-Afroman) [inline package.json:27]: are these spaces required?

**Review by Parsley-Afroman — COMMENTED** [review]
error with the PUT route of update item '/products/:SKU' = correct SKU code format for a non-existent db entity sends back a success response but with nothing actually done

File: Config/Routes.js (hunk 2 above)
Comment (luke-grayland) [reply Config/Routes.js:8]: Good shout, have done

**Pull request merged**